src/components/suggest/suggest_item.tsx

```diff
@@ -193,9 +193,9 @@ export const OuiSuggestItem: FunctionComponent<OuiSuggestItemProps> = ({
   if (onClick) {
     return (
       <button
-        onClick={onClick}
         className={classes}
-        {...(rest as PropsForButton)}>
+        {...(rest as PropsForButton)}
+        onClick={onClick}>
```

Comment:
Are you sure we should put `onClick` after `{{...(rest as PropsForButton)}`? I understand that TS gives an error there, but it would still be good to understand why it started giving it

Reply:
Typescript highlights that `onClick` is specified more than once, so usage will be overwritten

This is due usage of `as PropsForButton`, which reintroduces the `onClick` while it is already picked
```type PropsForButton = Omit<
  ButtonHTMLAttributes<HTMLButtonElement>,
  'onClick' | 'type'
> & {
  onClick: MouseEventHandler<HTMLButtonElement> | undefined;
};
```

Provided solution is easer than `{...(rest as Omit<PropsForButton, 'onClick'}`or refactoring of the `PropsForButton`

Reply:
Typing is chaged to suggested